data/maps/SP_Scenarios/Along_The_River.wmf/scripting/functions.lua

```diff
@@ -0,0 +1,119 @@
+-- =======================================================================
+--                             Along The River
+-- =======================================================================
+
+-- Checks objective for done flag.
+function obj_done(objective)
+   return (objective ~= nil and objective.done)
+end
+
+-- =======================================================================
+
+-- Places building at field if player owns field and extra field for flag.
+function place_building_if_owner(player, building, point, cs)
+   if point.owner == player and point.brn.owner == player then
```

Comment:
This will also work if the fields are directly on the player's border. Is this intended?

Reply:
Can't say for sure as this was written by fk. For this case I guess yes